src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,55 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new ValidationException("User can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new ValidationException("Login can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new ValidationException("Password can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new ValidationException("Age can't be null");
+        }
+        if (!isValidUserCredentials(user)) {
+            throw new ValidationException("Login and password should contain"
+                    + " at least 6 characters");
+        }
```

Comment:
The error message in this exception is not completely correct. This method checks both login and password length, but the error message only mentions 'Login and password'. It could be confusing for a user who has a correct login length but incorrect password length. It's better to split the validation of the login and password into two separate checks with their own error messages.